tools/sst_rewrite.cc

```diff
@@ -0,0 +1,353 @@
+// -------------------------------------------------------------------
+//
+// sst_rewrite.cc
+//
+// Copyright (c) 2015 Basho Technologies, Inc. All Rights Reserved.
+//
+// This file is provided to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file
+// except in compliance with the License.  You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+// -------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+//#include <libgen.h>
+
+#include "leveldb/env.h"
+#include "leveldb/filter_policy.h"
+#include "leveldb/options.h"
+#include "leveldb/table.h"
+#include "leveldb/table_builder.h"
+
+void command_help();
+
+// wrapper class for opening / closing existing leveldb tables
+class LDbTable
+{
+public:
+    LDbTable(leveldb::Options &, std::string &);
+    virtual ~LDbTable();
+
+    bool Ok() const {return(m_IsOpen);};
+    leveldb::Iterator * NewIterator();
+
+    const leveldb::Status & GetStatus() const {return(m_LastStatus);};
+    const char * GetFileName() const {return(m_FileName.c_str());};
+
+    uint64_t GetSstCounter(unsigned Idx) const
+        {return(m_IsOpen ? m_TablePtr->GetSstCounters().Value(Idx) : 0);};
+
+protected:
+    leveldb::Options & m_Options;
+    std::string m_FileName;
+    leveldb::RandomAccessFile * m_FilePtr;
+    leveldb::Table * m_TablePtr;
+    uint64_t m_FileSize;
+    leveldb::Status m_LastStatus;
+
+    bool m_IsOpen;
+
+private:
+    // disable these
+    LDbTable();
+    LDbTable(const LDbTable &);
+    const LDbTable operator=(const LDbTable&);
+};  // LDbTable
+
+
+LDbTable::LDbTable(
+    leveldb::Options & Options,
+    std::string & FileName)
+    : m_Options(Options), m_FileName(FileName),
+      m_FilePtr(NULL), m_TablePtr(NULL), m_FileSize(0), m_IsOpen(false)
+{
+    m_LastStatus=m_Options.env->GetFileSize(m_FileName, &m_FileSize);
+
+    if (m_LastStatus.ok())
+        m_LastStatus=m_Options.env->NewRandomAccessFile(m_FileName, &m_FilePtr);
```

Comment:
I prefer curly braces around single conditional statements: "if (bool) {statement;}" instead of "if (bool) statement;"

This avoids the potential bug of someone adding a second statement later and forgetting to wrap the two in braces.